lib/compiler.js

```diff
@@ -187,13 +186,13 @@ var importExportVisitor = new types.PathVisitor.fromMethodsObject({
         break;
       }
 
-      if (n.SwitchCase.check(stmtPath.value)) {
+      if (stmtPath.value.type === "SwitchCase") {
         assert.strictEqual(bodyPath.name, "consequent");
         insertCharIndex = bodyPath.value[0].start;
         break;
       }
 
-      if (n.Statement.check(stmtPath.value)) {
+      if (! Array.isArray(stmtPath.value)) {
```

Comment:
It's definitely not equivalent in general, but this change works because an `import` declaration can only be contained by a few kinds of nodes:
1. the `.body` array of a `Program`
2. the `.body` array of a `BlockStatement`
3. the `.consequent` array of a `SwitchCase`
4. a `Statement` without braces, such as:
```js
if (loggingEnabled) import "logging";
```
The general `Statement.check` API is something we gave up by removing `ast-types`, since it requires knowledge of all subtypes of `Statement`, which we don't have any more. But that's OK because we can still identify case 4 because it's the only case where this loop encounters an AST node before it encounters a `.body` array.

Of course, if you're using the top-level parser, this logic won't be exercised (in fact only the `Program` case should ever be reached), but Reify is intended to handle nested `import` declarations if you give it a full AST.